app/src/main/java/org/gdg/frisbee/android/view/LocationListPreference.java

```diff
@@ -0,0 +1,295 @@
+package org.gdg.frisbee.android.view;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.preference.DialogPreference;
+import android.preference.PreferenceManager;
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Adapter;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.Filter;
+import android.widget.ListView;
+
+import org.gdg.frisbee.android.R;
+
+public class LocationListPreference extends DialogPreference {
+
+    private FilterListView listView;
+    private int clickedItemIndex;
+    private String mValue;
+    private boolean mValueSet;
+    private String[] mEntries;
+    private String[] mEntryValues;
+
+    public LocationListPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDialogLayoutResource(R.layout.view_location_list_preference);
+    }
+
+    @Override
+    protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
+        super.onPrepareDialogBuilder(builder);
+        builder.setPositiveButton(null, null);
+    }
+
+    @Override
+    protected void onBindDialogView(View view) {
+        super.onBindDialogView(view);
+        listView = (FilterListView) view.findViewById(android.R.id.list);
+        clickedItemIndex = findIndexOfValue(mValue);
+        listView.setOnItemClickListener(
+                new AdapterView.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                        clickedItemIndex = findIndexByLabel((String) parent.getItemAtPosition(position));
```

Comment:
Actually on second thought, we may extend `ListPreference` instead of Dialog to get rid of all the List clicking functionality. Do you think it can be done? I am not sure though.